### PR TITLE
Update BBEs for Email Connector for Ballerina 2.0

### PR DESCRIPTION
**Pull request by Maninda**
## Purpose
Update BBEs related to Email Connector for Ballerina 2.0 release.

### CODE REVIEW

File: examples/send-and-receive-emails/send_and_receive_emails_4_listener.bal [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    // When an email is successfully received `onMessage` method is called
+    resource function onMessage(email:Email emailMessage) {
+        io:println("Email Subject: ", emailMessage.subject);
+        io:println("Email Body: ", emailMessage.body);
+    }
+
+    // When an error occurs during the email poll operations `onError` is called
+    resource function onError(email:Error emailError) {
+        io:println("Error while polling for the emails: "
+            + <string> emailError.message());
```
Comment (Bhashinee) [inline examples/send-and-receive-emails/send_and_receive_emails_4_listener.bal:36]:
```suggestion
            + emailError.message());
```

File: examples/send-and-receive-emails/send_and_receive_emails_3_imap.bal [hunk 2]
```diff
@@ -20,10 +20,10 @@ public function main() {
             io:println("There are no emails in the INBOX.");
         } else {
             io:println("Error while getting getting response: "
-                + <string> emailResponse.detail()["message"]);
+                + <string> emailResponse.message());
```
Comment (Bhashinee) [inline examples/send-and-receive-emails/send_and_receive_emails_3_imap.bal:23]:
```suggestion
                + emailResponse.message());
```

File: examples/send-and-receive-emails/send_and_receive_emails_1_smtp.bal [hunk 3]
```diff
@@ -37,7 +37,7 @@ public function main() {
     email:Error? response = smtpClient->send(email);
     if (response is email:Error) {
         io:println("Error while sending the email: "
-            + <string> response.detail()["message"]);
+            + <string> response.message();
```
Comment (Bhashinee) [inline examples/send-and-receive-emails/send_and_receive_emails_1_smtp.bal:40]:
```suggestion
            + response.message();
```
Do we need this casting? shall we check other places as well?

Comment (Bhashinee) [inline examples/send-and-receive-emails/send_and_receive_emails_1_smtp.bal:40]: Do we need this casting? Please check other places as well.
```suggestion
            + response.message();
```

File: examples/send-and-receive-emails/send_and_receive_emails_2_pop.bal [hunk 4]
```diff
@@ -20,10 +20,10 @@ public function main() {
             io:println("There are no emails in the INBOX.");
         } else {
             io:println("Error while getting getting response: "
-                + <string> emailResponse.detail()["message"]);
+                + <string> emailResponse.message());
         }
     } else {
         io:println("Error while creating client: "
-            + <string> popClient.detail()["message"]);
+            + <string> popClient.message());
```
Comment (kalaiyarasiganeshalingam) [inline examples/send-and-receive-emails/send_and_receive_emails_2_pop.bal:27]:
```suggestion
            + popClient.message());
```
Check all places

**Comment by kalaiyarasiganeshalingam** [comment]
Need to add the .metatags file

File: examples/send-and-receive-emails/send_and_receive_emails_2_pop.bal [hunk 5]
```diff
@@ -20,10 +20,10 @@ public function main() {
             io:println("There are no emails in the INBOX.");
         } else {
             io:println("Error while getting getting response: "
-                + <string> emailResponse.detail()["message"]);
+                + <string> emailResponse.message());
```
Comment (Bhashinee) [inline examples/send-and-receive-emails/send_and_receive_emails_2_pop.bal:23]:
```suggestion
                + emailResponse.message());
```

File: examples/send-and-receive-emails/send_and_receive_emails_3_imap.bal (hunk 2 above)
Comment (Bhashinee) [inline examples/send-and-receive-emails/send_and_receive_emails_3_imap.bal:23]:
```suggestion
                + emailResponse.message());
```

File: examples/send-and-receive-emails/send_and_receive_emails_3_imap.bal [hunk 6]
```diff
@@ -20,10 +20,10 @@ public function main() {
...
         }
     } else {
         io:println("Error while creating client: "
-            + <string> imapClient.detail()["message"]);
+            + <string> imapClient.message());
```
Comment (Bhashinee) [inline examples/send-and-receive-emails/send_and_receive_emails_3_imap.bal:27]:
```suggestion
            + imapClient.message());
```

**Pull request merged**